dynet/dim.h

```diff
@@ -212,6 +212,35 @@ struct Dim {
     }
   }
   /**
+   * \brief Remove multi-dimensions
+   * \param dims dimensions to be removed, assuming that there is no duplicate elements in it
+   * \param reduce_batch reduce the batch dimension or not
+   */
+  inline void delete_dims(std::vector<unsigned int> dims, bool reduce_batch){
+    DYNET_ARG_CHECK(dims.size() <= nd, "Out of bounds exception in Dim::delete_dims");
+    for(unsigned int i = 0; i < dims.size(); i++) {
+      DYNET_ARG_CHECK(dims[i] < nd, "Out of bounds exception in Dim::delete_dims");
```

Comment:
`DYNET_ARG_CHECK(!deleted_dims[dims[i]], "Duplicate dimension in Dim::delete_dims");`
`deleted_dims[dims[i]] = true`